projects/observable-webworker/src/lib/run-worker.ts

```diff
@@ -27,8 +27,6 @@ export function getWorkerResult<I, O>(
   const input$ = incomingMessages$.pipe(
     map((e: WorkerMessageNotification<I>): Notification<I> => e.data),
     map((n: Notification<I>) => new Notification(n.kind, n.value, n.error)),
-    // ignore complete, the calling thread will manage termination of the stream
-    filter(n => n.kind !== 'C'),
```

Comment:
this was clearly intentional, I need to get my head around why.

Reply:
We've been discussing with @zakhenry privately to understand why the demo worker with the files was working ok.
And Zak found the answer, I'll share here what he found so we remember and anyone interested can understand:

The [issue](https://github.com/cloudnc/observable-webworker/issues/116) reported by @christian-schlichtherle was happening when implementing `DoWork` as shown [here](https://github.com/cloudnc/observable-webworker/issues/116#issuecomment-2020481304).

But there's a split happening here:

https://github.com/cloudnc/observable-webworker/blob/2c9fe413bbba6f245d0399e9cea7eaebfb70a4b9/projects/observable-webworker/src/lib/run-worker.ts#L35-L37

And in the case of the `UnitType` the `materialize` was applied on the `from` :warning:. Therefore, when a `from` was done, it'd emit a complete event, even multiple time if the `concatMap` was kicking in several times. Hence why in this case it was receiving a complete and not in the `DoWork` one.

Still some investigation ongoing though with Zak to dig into the `materialize` atm.